feature-libs/commerce-quotes/core/facade/commerce-quotes.service.ts

```diff
@@ -1,59 +1,45 @@
 import { Injectable } from '@angular/core';
 import { ActiveCartFacade, MultiCartFacade } from '@spartacus/cart/base/root';
 import {
+  Comment,
   CommerceQuotesFacade,
   CommerceQuotesListReloadQueryEvent,
   Quote,
+  QuoteAction,
+  QuoteDetailsReloadQueryEvent,
   QuoteList,
   QuoteMetadata,
-  Comment,
-  QuoteAction,
+  QuoteStarter,
 } from '@spartacus/commerce-quotes/root';
 import {
   Command,
   CommandService,
   CommandStrategy,
   EventService,
-  GlobalMessageService,
-  GlobalMessageType,
+  LoginEvent,
   Query,
   QueryService,
   QueryState,
   RoutingService,
   UserIdService,
 } from '@spartacus/core';
-import { NavigationEvent, ViewConfig } from '@spartacus/storefront';
+import { ViewConfig } from '@spartacus/storefront';
 import { BehaviorSubject, combineLatest, Observable, of, zip } from 'rxjs';
 import {
-  switchMap,
-  withLatestFrom,
-  take,
   concatMap,
   map,
+  switchMap,
+  take,
   tap,
+  withLatestFrom,
 } from 'rxjs/operators';
 import { CommerceQuotesConnector } from '../connectors/commerce-quotes.connector';
 
 @Injectable()
 export class CommerceQuotesService implements CommerceQuotesFacade {
   protected currentPage$ = new BehaviorSubject<number>(0);
   protected sortBy$ = new BehaviorSubject<string>('byCode');
-
-  protected quotesState$: Query<QuoteList, unknown[]> =
-    this.queryService.create<QuoteList>(
-      () =>
-        this.userIdService.takeUserId().pipe(
-          withLatestFrom(this.currentPage$, this.sortBy$),
-          switchMap(([userId, currentPage, sort]) =>
-            this.commerceQuotesConnector.getQuotes(userId, {
-              currentPage,
-              sort,
-              pageSize: this.config.view?.defaultPageSize,
-            })
-          )
-        ),
-      { reloadOn: [CommerceQuotesListReloadQueryEvent] }
-    );
+  protected isLoading$ = new BehaviorSubject<boolean>(false);
```

Comment:
I'm not sure if I'm a fan of this solution. We're putting information for a single component in general service. In this case, we should have a separate service as it was done in `CommerceQuotesActionLinksService` - a dedicated service for such cases to avoid putting contextual data into general services.

Reply:
Yep, I agree it should be moved to separate component service 👍

Reply:
Well, it turned out, that it cannot be moved to component-specific service, as this flag probably should be set each time some action is performed (not only requote) to inform user that something is loading under the hood. We can discuss whether there is a different approach to handle such case.

Reply:
Finally it stays after discussion we had. Flag is necessary for all CMS-driven components.